x-pack/plugins/apm/common/rules/schema.ts

```diff
@@ -15,6 +15,7 @@ export const errorCountParamsSchema = schema.object({
   threshold: schema.number(),
   serviceName: schema.maybe(schema.string()),
   environment: schema.string(),
+  groupBy: schema.maybe(schema.arrayOf(schema.string())),
```

Comment:
in which cases the `groupBy` can be undefined?

Reply:
For the existing rules, the `groupBy` will be undefined. If Kibana is upgraded with this PR when there are existing APM rules running, `groupBy` will be undefined.

Reply:
yes, the executer checks whether `groupBy` is there. If not, it will add preSelected fields.
```
const allGroupbyFields: string[] = [
  ruleParams.groupBy ?? predefinedGroupby,
].flat();
```